cmd/lifecycle/exporter.go

```diff
@@ -221,9 +234,29 @@ func (e *exportCmd) export(group buildpack.Group, cacheStore lifecycle.Cache, an
 }
 
 func (e *exportCmd) initDaemonAppImage(analyzedMD platform.AnalyzedMetadata) (imgutil.Image, string, error) {
+	if isDigestRef(e.RunImageRef) {
+		// If the run image reference is a digest reference, this means extensions were used to extend the runtime base image.
```

Comment:
this comment seems like it might not age well since there could be other reasons in the future to have digest references,  but it's also surprising to me that we're guaranteed to turn a digest back into a tag?   also it seems like the implementation of parseReference in GGCR will actually 'succeed' if it just turns it back into a digest?

```
// ParseReference parses the string as a reference, either by tag or digest.
func ParseReference(s string, opts ...Option) (Reference, error) {
	if t, err := NewTag(s, opts...); err == nil {
		return t, nil
	}
	if d, err := NewDigest(s, opts...); err == nil {
		return d, nil
	}
	return nil, newErrBadName("could not parse reference: " + s)
}
```

so maybe i'm not fully appreciating what we're up to here.   but i think mostly it's just the comment that seems like it's not future proof. e.g. rewording it as
`// if extensions were used to extend the image them the run image reference will be a digest reference`
fixes that fragility, b/c it means if something else comes along in the future that also has the same result, that sentence will still be true.   (whereas the current wording implies that run image extensions is the only process that could have this result)

Reply:
Fair point, updated the comment in f24dc342